tests/components/smartthings/test_config_flow.py

```diff
@@ -224,128 +283,159 @@ async def test_cloudhook_app_created_then_show_wait_form(
     flow = SmartThingsFlowHandler()
     flow.hass = hass
 
     smartthings_mock.apps.return_value = [app]
     smartthings_mock.generate_app_oauth.return_value = app_oauth_client
+    smartthings_mock.locations.return_value = [location]
+    token = str(uuid4())
 
-    result = await flow.async_step_user({"access_token": str(uuid4())})
+    result = await flow.async_step_pat({CONF_ACCESS_TOKEN: token})
 
+    assert flow.access_token == token
+    assert flow.app_id == app.app_id
+    assert flow.oauth_client_secret == app_oauth_client.client_secret
+    assert flow.oauth_client_id == app_oauth_client.client_id
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["step_id"] == "wait_install"
+    assert result["step_id"] == "select_location"
 
 
-async def test_wait_form_displayed(hass):
-    """Test the wait for installation form is displayed."""
+async def test_step_pat_app_updated_webhook_from_existing_oauth_client(
+    hass, app, location, smartthings_mock
+):
+    """Test SmartApp is updated from existing then show location form."""
+    oauth_client_id = str(uuid4())
+    oauth_client_secret = str(uuid4())
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_APP_ID: app.app_id,
+            CONF_OAUTH_CLIENT_ID: oauth_client_id,
+            CONF_OAUTH_CLIENT_SECRET: oauth_client_secret,
+            CONF_LOCATION_ID: str(uuid4()),
+        },
+    )
+    entry.add_to_hass(hass)
     flow = SmartThingsFlowHandler()
     flow.hass = hass
+    smartthings_mock.apps.return_value = [app]
+    smartthings_mock.locations.return_value = [location]
+    token = str(uuid4())
 
-    result = await flow.async_step_wait_install(None)
+    result = await flow.async_step_pat({CONF_ACCESS_TOKEN: token})
 
+    assert flow.access_token == token
+    assert flow.app_id == app.app_id
+    assert flow.oauth_client_secret == oauth_client_secret
+    assert flow.oauth_client_id == oauth_client_id
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["step_id"] == "wait_install"
+    assert result["step_id"] == "select_location"
 
 
-async def test_wait_form_displayed_after_checking(hass, smartthings_mock):
-    """Test error is shown when the user has not installed the app."""
+async def test_step_select_location(hass, location, smartthings_mock):
+    """Test select location shows form with available locations."""
+    smartthings_mock.locations.return_value = [location]
     flow = SmartThingsFlowHandler()
     flow.hass = hass
-    flow.access_token = str(uuid4())
-
-    result = await flow.async_step_wait_install({})
+    flow.api = smartthings_mock
 
+    result = await flow.async_step_select_location()
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["step_id"] == "wait_install"
-    assert result["errors"] == {"base": "app_not_installed"}
+    assert result["step_id"] == "select_location"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5359319)